From 1706118978968704a6a97ab503401683081db050 Mon Sep 17 00:00:00 2001
From: Eric Yang <eric.yang2@amd.com>
Date: Fri, 8 Jul 2016 17:16:23 -0400
Subject: [PATCH 0614/1722] drm/amd/dal: make rect offset signed int

This fixes the hang when video plane has negative offsets
Also add print to print out clip rect

Change-Id: I291e09514a09c19c19769254db4138081bfce0e1
Signed-off-by: Eric Yang <eric.yang2@amd.com>
Acked-by: Harry Wentland <harry.wentland@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/dal/dc/dc_hw_types.h                | 4 ++--
 drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c | 9 +++++++--
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/dc/dc_hw_types.h b/drivers/gpu/drm/amd/dal/dc/dc_hw_types.h
index 7c0879e..2ed6280 100644
--- a/drivers/gpu/drm/amd/dal/dc/dc_hw_types.h
+++ b/drivers/gpu/drm/amd/dal/dc/dc_hw_types.h
@@ -82,8 +82,8 @@ struct dc_plane_address {
 };
 
 struct rect {
-	uint32_t x;
-	uint32_t y;
+	int x;
+	int y;
 	uint32_t width;
 	uint32_t height;
 };
diff --git a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
index bdca0dc..64633e9 100644
--- a/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
+++ b/drivers/gpu/drm/amd/dal/dc/dce110/dce110_hw_sequencer.c
@@ -1878,7 +1878,8 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 			"Pipe:%d 0x%x: addr hi:0x%x, "
 			"addr low:0x%x, "
 			"src: %d, %d, %d,"
-			" %d; dst: %d, %d, %d, %d;\n",
+			" %d; dst: %d, %d, %d, %d;"
+			"clip: %d, %d, %d, %d\n",
 			pipe_ctx->pipe_idx,
 			pipe_ctx->surface,
 			pipe_ctx->surface->public.address.grph.addr.high_part,
@@ -1890,7 +1891,11 @@ static void dce110_program_front_end_for_pipe(struct core_dc *dc,
 			pipe_ctx->surface->public.dst_rect.x,
 			pipe_ctx->surface->public.dst_rect.y,
 			pipe_ctx->surface->public.dst_rect.width,
-			pipe_ctx->surface->public.dst_rect.height);
+			pipe_ctx->surface->public.dst_rect.height,
+			pipe_ctx->surface->public.clip_rect.x,
+			pipe_ctx->surface->public.clip_rect.y,
+			pipe_ctx->surface->public.clip_rect.width,
+			pipe_ctx->surface->public.clip_rect.height);
 
 
 	dal_logger_write(dc->ctx->logger,
-- 
2.7.4

